post-start.d/62_call_logd_tenant_migration.py

```diff
@@ -0,0 +1,143 @@
+    'config_file': '/etc/wazo-upgrade/config.yml',
+    'auth': {
+        'key_file': '/var/lib/wazo-auth-keys/wazo-upgrade-key.yml'
+    }
+}
+
+requests.packages.urllib3.disable_warnings(category=InsecureRequestWarning)
+
+
+def _load_config():
+    file_config = read_config_file_hierarchy(_DEFAULT_CONFIG)
+    key_config = _load_key_file(ChainMap(file_config, _DEFAULT_CONFIG))
+    return ChainMap(key_config, file_config, _DEFAULT_CONFIG)
+
+
+def _load_key_file(config):
+    key_file = parse_config_file(config['auth']['key_file'])
+    return {
+        'auth': {'username': key_file['service_id'], 'password': key_file['service_key']},
+    }
+
+
+def _wait_for_call_logd(call_logd_config):
+    url = 'https://{host}:{port}/{version}/status'.format(**call_logd_config)
+    for _ in range(30):
+        try:
+            requests.get(url, verify=False)
+            return
+        except requests.exceptions.ConnectionError:
+            time.sleep(1.0)
+
+    print('call_logd tenant migration failed, could not connect to wazo-call-logd')
+    sys.exit(2)
+
+
+@contextmanager
+def _migration_plugin(call_logd_config):
+
+    config_filename = '/etc/wazo-call-logd/conf.d/50-wazo-tenant-migration.yml'
+    config_file = '''\
+enabled_plugins:
+  tenant_migration: true
+'''
+
+    with open(config_filename, 'w') as f:
+        f.write(config_file)
+    subprocess.run(['systemctl', 'restart', 'wazo-call-logd'])
+    _wait_for_call_logd(call_logd_config)
+
+    try:
+        yield
+    finally:
+        try:
+            os.unlink(config_filename)
+        except OSError:
+            pass
+        subprocess.run(['systemctl', 'restart', 'wazo-call-logd'])
+
+
+def migrate_tenants():
+    config = _load_config()
+    auth_client = AuthClient(**config['auth'])
+    token = auth_client.token.new('wazo_user', expiration=5 * 60)['token']
+    auth_client.set_token(token)
+    confd = ConfdClient(token=token['token'], **config['confd'])
+
+    body = {
+        'contexts': [
+            {"context": context['name'], "tenant_uuid": context['tenant_uuid']}
+            for context in confd.contexts.list(recurse=True)["items"]
+        ]
+    }
+
+    with _migration_plugin(config['call_logd']):
+        url = 'https://{host}:{port}/{version}/tenant-migration'.format(**config['call_logd'])
+        result = requests.post(
+            url,
+            data=json.dumps(body),
+            headers={'X-Auth-Token': token, 'Content-Type': 'application/json'},
+            verify=False,
+        )
+
+        if result.status_code != 200:
+            print(
+                'call_logd tenant migration failed, status-code {}:\n'
+                '{}\ncheck /var/log/wazo-call-logd.log for more info'
+                .format(result.status_code, result.text)
+            )
+            sys.exit(2)
+
+
+def main():
+    args = parse_args()
+
+    if not args.force:
+        version_installed = os.getenv('XIVO_VERSION_INSTALLED')
+        if version_installed > '19.09':
```

Comment:
```suggestion
        if version_installed > '19.10':
```

Reply:
I forget that because I wrote that in previous sprint ...